File: BEPiD/BepidAulas/aula22/NotificationObjC/NotificationObjC/ViewController2.m
//
//  ViewController2.m
//  NotificationObjC
//
//  Created by Danilo Barros Mendes on 5/4/15.
//  Copyright (c) 2015 Danilo Barros Mendes. All rights reserved.
//

#import "ViewController2.h"

@interface ViewController2 ()

@end

@implementation ViewController2

-(instancetype)initWithNibName: (NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil{
    
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    
    if (self) {
        
        self.tabBarItem.title = @"View2";
        self.tabBarItem.image = [UIImage imageNamed:@"Time.png"];
        
    }
    
    return self;
    
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _date.minimumDate = [NSDate dateWithTimeIntervalSinceNow:60];
    
    
    
}
-(void)viewWillAppear:(BOOL)animated{
    self.tabBarItem.badgeValue = nil;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)notificationAlert:(UIButton *)sender {
    
    UILocalNotification *notification = [[UILocalNotification alloc] init];
    
    notification.alertBody = @"Minha notificacao";
    notification.fireDate = _date.date;
    
    [[UIApplication sharedApplication] scheduleLocalNotification:notification];
    
    NSLog(@"Notificacao");
    
    
}

- (IBAction)AddTab:(id)sender {
    
    NSMutableArray *array = [[NSMutableArray alloc] initWithArray:self.tabBarController.viewControllers];
    
    UIViewController *new = [UIViewController new];
    
    new.tabBarItem = [[UITabBarItem alloc] initWithTitle:@"New"
                                                   image:[UIImage imageNamed:@"Time.png"]
                                                     tag:[self.tabBarController.viewControllers count]];
    
    [array addObject:new];
    
    
    [self.tabBarController setViewControllers:array animated:YES];
    
    self.tabBarItem.badgeValue = [NSString stringWithFormat:@"%d",[self.tabBarItem.badgeValue intValue]+1];
    
    //NSLog(@"%lu", (unsigned long)[self.tabBarController.viewControllers count]);
    
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: BEPiD/BepidAulas/AulaViewController/AulaViewController/ViewController2.swift
//
//  ViewController2.swift
//  AulaViewController
//
//  Created by Danilo Barros Mendes on 5/4/15.
//  Copyright (c) 2015 Danilo Barros Mendes. All rights reserved.
//

import UIKit

class ViewController2: UIViewController {
    
    @IBOutlet weak var datePicker: UIDatePicker!
    
    @IBAction func addNotification(sender: AnyObject){
        let date = datePicker.date
        println("Agendado notificação para \(date)")
        
//        let today = NSDate()
//        
//        if date.compare(today) == NSComparisonResult.OrderedAscending{
//            
//        }else{
//            
//        }
//        
//        

        let notification = UILocalNotification()
        notification.alertBody = "Minha notificacao"
        notification.fireDate = date
        UIApplication.sharedApplication().scheduleLocalNotification(notification)
        
        
    }
    
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?){
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
        tabBarItem.title = "NotificationView"
        tabBarItem.image = UIImage(named: "Time.png")
    }
    
    convenience init (){
        self.init(nibName: nil, bundle: nil)
    }
    
    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        
        datePicker.minimumDate = NSDate(timeIntervalSinceNow: 60);
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        println("Carregou view1")
        
        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
